File: practice/python/number_palindrome_or_not.py
def number_palindrome_or_not(number):
  
    reverse = 0;
    original = number
    while(number != 0):
        digit = number % 10
        reverse = (reverse * 10) + digit
        number =  number // 10
    if(original == reverse ):
        print("number is a palindrome")
    else:
       print("number is a  not palindrome")
                                  


print("enter a number")
output = input()
number = int(output)
number_palindrome_or_not(number)

File: practice/java/NumberPalindromeOrNot.java

import java.util.Scanner;

class NumberPalindromeOrNot {

    static void numberPalindeome(int number) {
        int orignal = number;
        int digit;
        int reverse = 0;
        while (number != 0) {
            digit = number % 10;
            reverse = (reverse * 10) + digit;
            number = number / 10;
        }
        if (orignal == reverse) {
            System.out.println("number is palindarome");

        } else {
            System.out.println("number is not palindarome");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("enter a number:");
        int number = scanner.nextInt();
        numberPalindeome(number);
    }
}
